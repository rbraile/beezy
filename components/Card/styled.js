import styled from 'styled-components'

export const CardContainer = styled.article`
  background: #eee;
  border: 1px solid #000;
  border-radius: 5px;
  padding: 10px;
  width: 33%;
  margin-bottom: 15px;
  display: flex;
`

export const DescriptionContainer = styled.div`
  margin-left: 10px;
`

export const Name = styled.h2`
  font-size: 18px;
  font-weight: bold;
  color: #094d8c;
`
