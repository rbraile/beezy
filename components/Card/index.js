import React from 'react'
import Link from 'next/link'
import { CardContainer, DescriptionContainer, Name } from './styled'
import Fields from '../Fields'

const Card = character => {
  const id = character.url.split('people/')[1].replace('/', '')
  const detailList = [
    'gender',
    'height',
    'hair_color',
    'skin_color',
    'eye_color',
    'birth_year',
    'mass',
  ]
  return (
    <CardContainer key={`people-${id}`}>
      <Link href={`/detail?id=${id}`}>
        <DescriptionContainer>
          <Name>{character.name}</Name>
          <Fields detailList={detailList} details={character} />
        </DescriptionContainer>
      </Link>
    </CardContainer>
  )
}

export default Card
