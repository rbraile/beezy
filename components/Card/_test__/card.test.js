import React from 'react'
import { mount, shallow } from 'enzyme'

import Card from '../index'

describe('TEST = CARD', () => {
  const character = {
    name: 'Anakin Skywalker',
    height: '188',
    mass: '84',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '41.9BBY',
    gender: 'male',
    url: 'https://swapi.co/api/people/1/',
  }

  test('TEST = CARD = SNAPSHOT', () => {
    const wrapper = mount(<Card {...character} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('TEST = CARD = Has the 87 paragraph one for each detail', () => {
    const wrapper = mount(<Card {...character} />)
    expect(wrapper.find('p')).toHaveLength(7)
  })
})
