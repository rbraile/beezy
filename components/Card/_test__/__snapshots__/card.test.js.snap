// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST = CARD TEST = CARD = SNAPSHOT 1`] = `
<Card
  birth_year="41.9BBY"
  eye_color="blue"
  gender="male"
  hair_color="blond"
  height="188"
  mass="84"
  name="Anakin Skywalker"
  skin_color="fair"
  url="https://swapi.co/api/people/1/"
>
  <styled__CardContainer
    key="people-1"
  >
    <article
      className="styled__CardContainer-sc-4po4d9-0 cPRHzo"
    >
      <Link
        href="/detail?id=1"
        prefetch={true}
      >
        <styled__DescriptionContainer
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <a
            className="styled__DescriptionContainer-sc-4po4d9-1 djNYan"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <styled__Name>
              <h2
                className="styled__Name-sc-4po4d9-2 dsMHFm"
              >
                Anakin Skywalker
              </h2>
            </styled__Name>
            <Fields
              detailList={
                Array [
                  "gender",
                  "height",
                  "hair_color",
                  "skin_color",
                  "eye_color",
                  "birth_year",
                  "mass",
                ]
              }
              details={
                Object {
                  "birth_year": "41.9BBY",
                  "eye_color": "blue",
                  "gender": "male",
                  "hair_color": "blond",
                  "height": "188",
                  "mass": "84",
                  "name": "Anakin Skywalker",
                  "skin_color": "fair",
                  "url": "https://swapi.co/api/people/1/",
                }
              }
            >
              <p
                key="gendermale"
              >
                <span>
                  gender
                  : 
                </span>
                <strong>
                  male
                </strong>
              </p>
              <p
                key="height188"
              >
                <span>
                  height
                  : 
                </span>
                <strong>
                  188
                </strong>
              </p>
              <p
                key="hair_colorblond"
              >
                <span>
                  hair_color
                  : 
                </span>
                <strong>
                  blond
                </strong>
              </p>
              <p
                key="skin_colorfair"
              >
                <span>
                  skin_color
                  : 
                </span>
                <strong>
                  fair
                </strong>
              </p>
              <p
                key="eye_colorblue"
              >
                <span>
                  eye_color
                  : 
                </span>
                <strong>
                  blue
                </strong>
              </p>
              <p
                key="birth_year41.9BBY"
              >
                <span>
                  birth_year
                  : 
                </span>
                <strong>
                  41.9BBY
                </strong>
              </p>
              <p
                key="mass84"
              >
                <span>
                  mass
                  : 
                </span>
                <strong>
                  84
                </strong>
              </p>
            </Fields>
          </a>
        </styled__DescriptionContainer>
      </Link>
    </article>
  </styled__CardContainer>
</Card>
`;
